exploration/selection-matching-options.md

```diff
@@ -0,0 +1,547 @@
+* **Resource fallback**, particularly with sparsely populated localized resource files
+* **Skeletons** such as for dates (for example, `yyyyMMddjm`), which do not require translators to touch "picture strings" (such as `MM/dd/yyyy HH:mm a`) to handle the time or date separators `/` and `:`, the use of 24-hour time vs. 12-hour time, the order of the fields. 
+* **Built-in formats** such as `short`/`medium`/`long`/`full` do not guarantee any particular separators, field order, or format and vary widely between locales.
+* **Locale negotiation** matches the best particular locale to a requested language range.
+
+When coding a plural using ICU4J's `PluralFormat`, the developer only needs to worry about _specific value_ messages (`when 1 {This is your last chance.}`) vs. value based messages (`when one {You have {$count} chance remaining.}`).
+
+Many programming languages feature multi-value selection using logic constructs such as Java's `switch` statement. These statements do depend on the order of the "cases":
+
+```java
+switch (count) {
+   case 0:
+   case 1:
+   case 2:
+   default:
+};
+```
+
+Programming constructs like `switch` generally match a single value exactly. Sometimes developers use features, such as "fall through", to allow two or more blocks of code to operate on a single value. But the core of a `switch` is binary matching of the value. Generally the same input (`count` in the example above) can't match more than one `case`.
+
+We could model `match` as a switch:
+
+```
+match {$count}
+when 1   {This is your last chance}
+when one {You have {$count} chance remaining}
+when *   {...}
+```
+vs.
+```
+match {$count}
+when one {You have {$count} chance remaining}
+when 1   {This message cannot be reached in English, but would be reached in Japanese!!}
+when *   {...}
+```
+
+This exposes developers and translators to managing the complexity versus having the API take care of it.
+
+
+#### Are there other complex matching cases? Or is `plural` everything?
+
+Currently there are no other complex rule-based selectors in ICU. However, there are a number of cases where complex matching might come into play. The criteria for it being a complex match are:
+
+1. The selector generates different output by locale (the way that plural generates `few` and `many` for e.g. Polish or fails to generate `one` in Japanese). If the states vary according to something other than locale all of the states have to be accounted for in the `root` locale resource and all translations (so it's not "complex" any more); or:
```

Comment:
I don't really understand this use-case. How does this result in complex matching?